Firebase/Auth/Source/FIRAuth.m

```diff
@@ -676,6 +676,45 @@ - (void)signInWithCredential:(FIRAuthCredential *)credential
   }];
 }
 
+- (void)signInAnonymouslyAndRetrieveDataWithCompletion:(FIRAuthDataResultCallback)completion {
+  dispatch_async(FIRAuthGlobalWorkQueue(), ^{
+    FIRAuthDataResultCallback decoratedCallback =
+        [self signInFlowAuthDataResultCallbackByDecoratingCallback:completion];
+    if (_currentUser.anonymous) {
+      FIRAdditionalUserInfo *additionalUserInfo =
+          [[FIRAdditionalUserInfo alloc] initWithProviderID:nil
+                                                    profile:nil
+                                                   username:nil
+                                                  isNewUser:NO];
+      decoratedCallback([[FIRAuthDataResult alloc] initWithUser:_currentUser
+                                             additionalUserInfo:additionalUserInfo],
```

Comment:
Extract the FIRAuthDataResult as a local var.

Reply:
I thought of doing this, it's certainly cleaner.